server/src/test/java/io/druid/segment/indexing/DataSchemaTest.java

```diff
@@ -211,4 +211,89 @@ public void testSerde() throws Exception
         actual
     );
   }
+
+
+  @Test
+  public void testEquals() throws Exception
+  {
+    String jsonStr = "{"
+                     + "\"dataSource\":\"test\","
+                     + "\"parser\":{"
+                     + "\"type\":\"string\","
+                     + "\"parseSpec\":{"
+                     + "\"format\":\"json\","
+                     + "\"timestampSpec\":{\"column\":\"xXx\", \"format\": \"auto\", \"missingValue\": null},"
+                     + "\"dimensionsSpec\":{\"dimensions\":[\"foo\"], \"dimensionExclusions\":[]},"
+                     + "\"flattenSpec\":{\"useFieldDiscovery\":true, \"fields\":[]},"
+                     + "\"featureSpec\":{}},"
+                     + "\"encoding\":\"UTF-8\""
+                     + "},"
+                     + "\"metricsSpec\":[{\"type\":\"doubleSum\",\"name\":\"metric1\",\"fieldName\":\"col1\"}],"
+                     + "\"granularitySpec\":{"
+                     + "\"type\":\"arbitrary\","
+                     + "\"queryGranularity\":{\"type\":\"duration\",\"duration\":86400000,\"origin\":\"1970-01-01T00:00:00.000Z\"},"
+                     + "\"intervals\":[\"2014-01-01T00:00:00.000Z/2015-01-01T00:00:00.000Z\"]}}";
+
+    DataSchema actual = jsonMapper.readValue(
+        jsonMapper.writeValueAsString(
+            jsonMapper.readValue(jsonStr, DataSchema.class)
```

Comment:
no reason, it was just copied from the serde test above it, I'll get rid of that if this is a viable approach